### PR TITLE
kz _attributes.erl: obsolete phone numbers search

### PR DESCRIPTION
**Pull request by onnet**

### CODE REVIEW

File: core/kazoo_endpoint/src/kz_attributes.erl [hunk 1]
```diff
@@ -265,27 +265,28 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
...
+            {Number, Name}
+    end;
+maybe_get_assigned_number(_Number, Name, Call) ->
+    AccountDb = kapps_call:account_db(Call),
+    maybe_get_assigned_number(_Number, Name, AccountDb).
 
--spec maybe_get_assigned_numbers(ne_binaries(), ne_binary(), kapps_call:call()) ->
+-spec maybe_get_assigned_numbers(ne_binaries(), ne_binary(), ne_binary()|kapps_call:call()) ->
                                         {api_binary(), api_binary()}.
 maybe_get_assigned_numbers([], Name, _) ->
```
Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: Since last arg is unused, mind dropping it?

File: core/kazoo_endpoint/src/kz_attributes.erl [hunk 2]
```diff
@@ -265,27 +265,28 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
...
-maybe_get_assigned_number(_, Name, Call) ->
-    AccountDb = kapps_call:account_db(Call),
-    case kz_datamgr:open_cache_doc(AccountDb, ?KNM_PHONE_NUMBERS_DOC) of
-        {'error', _R} ->
-            Number = default_cid_number(),
-            lager:warning("could not open ~s doc <~s> ~s: ~p", [?KNM_PHONE_NUMBERS_DOC, Name, Number, _R]),
-            {Number, Name};
-        {'ok', JObj} ->
-            PublicJObj = kz_json:public_fields(JObj),
+maybe_get_assigned_number(_, Name, <<AccountDb/binary>>) ->
```
Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: I'd rather you use MATCH_ACCOUNT_ENCODED

**Comment by onnet** [comment]
@fenollp 
By the way, imho it loks like there is another inconcistency in kz_attributes.erl

It looks for <<"kz_attributes/owner">>
https://github.com/2600hz/kazoo/blob/master/core/kazoo_endpoint/src/kz_attributes.erl#L418

But https://github.com/2600hz/kazoo/blob/master/applications/callflow/priv/couchdb/views/attributes.json
provides just <<"attributes/owner">>

**Comment by jamhed** [comment]
@onnet https://github.com/2600hz/kazoo/pull/2283

File: core/kazoo_endpoint/src/kz_attributes.erl [hunk 3]
```diff
@@ -265,33 +265,33 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
...
+maybe_get_assigned_number(_, Name, ?MATCH_ACCOUNT_ENCODED(_)=AccountDb) ->
+    case knm_numbers:account_listing(AccountDb) of
+        [_|_] = NumbersList ->
             Numbers = [Num
-                       || Num <- kz_json:get_keys(PublicJObj)
+                       || {Num,JObj} <- NumbersList
                               ,Num =/= <<"id">>
-                              ,(not kz_json:is_true([Num, <<"on_subaccount">>], JObj))
-                              ,(kz_json:get_value([Num, <<"state">>], JObj) =:= ?NUMBER_STATE_IN_SERVICE)
+                              ,(not kz_json:is_true(<<"on_subaccount">>, JObj))
```
Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: This does not appear in the view. Maybe add it to ./applications/crossbar/priv/couchdb/account/phone_numbers.json ?

Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: This does not appear in the view. Maybe add it to ./applications/crossbar/priv/couchdb/account/phone_numbers.json ?

**Comment by onnet** [comment]
Hi @fenollp 

I can't find this flag mentioned anywhere else in the code.
So it looks like it reserved for the future use which make sense or a kind of an atavism.
Therefore it is up to you guys to make this decision. 
Let me know what you think and I add/delete it.

Regards,

**Comment by fenollp** [comment]
@onnet yes this is very strange. Could you replace that JSON field check with a check of whether `assigned_to` is defined?

(found occurrences of `on_subaccount` on more than 3 years old commits…)

File: core/kazoo_endpoint/src/kz_attributes.erl [hunk 4]
```diff
@@ -265,33 +265,33 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
...
+maybe_get_assigned_number(_, Name, ?MATCH_ACCOUNT_ENCODED(_)=AccountDb) ->
+    case knm_numbers:account_listing(AccountDb) of
+        [_|_] = NumbersList ->
             Numbers = [Num
-                       || Num <- kz_json:get_keys(PublicJObj)
+                       || {Num,JObj} <- NumbersList
                               ,Num =/= <<"id">>
-                              ,(not kz_json:is_true([Num, <<"on_subaccount">>], JObj))
-                              ,(kz_json:get_value([Num, <<"state">>], JObj) =:= ?NUMBER_STATE_IN_SERVICE)
+                              ,(kz_json:get_value(<<"assigned_to">>, JObj) =:= kz_util:format_account_id(AccountDb, 'raw'))
```
Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: Could you put `AccountId = kz_util:format_account_id(AccountDb)` outside of the LC loop?

File: core/kazoo_endpoint/src/kz_attributes.erl [hunk 5]
```diff
@@ -265,33 +265,33 @@ maybe_get_account_default_number(Number, Name, Account, Call) ->
...
-maybe_get_assigned_number(_, Name, Call) ->
-    AccountDb = kapps_call:account_db(Call),
-    case kz_datamgr:open_cache_doc(AccountDb, ?KNM_PHONE_NUMBERS_DOC) of
-        {'error', _R} ->
-            Number = default_cid_number(),
-            lager:warning("could not open ~s doc <~s> ~s: ~p", [?KNM_PHONE_NUMBERS_DOC, Name, Number, _R]),
-            {Number, Name};
-        {'ok', JObj} ->
-            PublicJObj = kz_json:public_fields(JObj),
+maybe_get_assigned_number(_, Name, ?MATCH_ACCOUNT_ENCODED(_)=AccountDb) ->
```
Comment (fenollp) [inline core/kazoo_endpoint/src/kz_attributes.erl]: I just noticed the first argument is always ignored. Could you fix this also?

**Comment by fenollp** [comment]
Thanks for your work & patience @onnet! I added a couple comments that slipped me. I can fix that myself if you want.

**Comment by onnet** [comment]
@fenollp No worries, will do in a couple of minutes..

**Pull request merged**